File: reference.net/db4o-reference-chapters/src/com/db4odoc/enums/EnumSODAExample.cs
/* Copyright (C) 2004 - 2006 db4objects Inc. http://www.db4o.com */

using System;
using Db4objects.Db4o;
using Db4objects.Db4o.Query;

namespace Db4objects.Db4odoc.Enums
{
	class EnumSODAExample 
	{
		public static IObjectSet FindOpenDoors(IObjectContainer container) 
		{
			IQuery query = container.Query();
			query.Constrain(typeof(Door));
			query.Descend("_state").Constrain(DoorState.Open);
			return query.Execute();
		}
	}
}

File: reference.vb.net/Db4objects.Db4odoc/Enums/EnumSODAExample.vb
' Copyright (C) 2004 - 2006 db4objects Inc. http://www.db4o.com 
Imports System
Imports Db4objects.Db4o
Imports Db4objects.Db4o.Query

Namespace Db4objects.Db4odoc.Enums
    Public Class EnumSODAExample
        Public Shared Function FindOpenDoors(ByVal container As IObjectContainer) As IObjectSet
            Dim query As IQuery = container.Query()
            query.Constrain(GetType(Door))
            query.Descend("_state").Constrain(DoorState.Open)
            Return query.Execute()
        End Function
    End Class
End Namespace
	